### DISCUSSION TITLE
Activity gets logged twice

### DISCUSSION THREAD

**Discussion by astarialexi** [spatie/laravel-activitylog#754] [Q&A] [answered]
So my activity is getting logged twice on the belongsTo relationship. I'm using [laravel backpack](https://backpackforlaravel.com/).

To explain further. For example I have a User and a User Type model. So User will have a user_type_id as field. So from my understanding the relationship should be User belongsTo UserType since User is the one that has the foreign key of User Type.

So i have this code on my App\User.php file:

```php
public function user(){
    return $this->belongsTo('App\UserType', 'user_type_id', 'id');
}
```

Then I have the logging code on App\Models\BackpackUser.php
```php
public function save(array $options = [])
{
    if (empty($this->id)) {
        parent::save($options);
        activity('user')
            ->causedBy($this)
            ->performedOn(backpack_user())
            ->log('created');
    } else {
        parent::save($options);
        activity('user')
            ->causedBy($this)
            ->performedOn(backpack_user())
            ->log('updated');
    }
}
```

And when I create/edit User, the activity gets logged twice. All field has the same values, even the timestamps. But when I change the relationship to hasOne, it only gets logged once.

Am I doing something wrong?

**Comment by Gummibeer** [comment, marked as answer]
Hey,

do you use the package traits on any model?
https://docs.spatie.be/laravel-activitylog/v3/advanced-usage/logging-model-events/
I highly recommend to do so, because they keep track on model events by default and will create a log automatically.

I don't know backpack, but could it be that backpack uses the activitylog by default? And your custom log is the duplicate?

Your `belongsTo` relationship is right - the model with the foreign ID in it's own table is the one that belongs to the model with only its primary ID.
https://laravel.com/docs/7.x/eloquent-relationships#one-to-many

```
user:
- id
- user_type_id
- email

user_type:
- id
- type
```

The `User` model belongs to the `UserType` model and the `UserType` model has many `User` models.

**Reply by astarialexi** [reply]
yeah we installed this: https://docs.spatie.be/laravel-activitylog/v3/introduction/
but I don't use it like how you said. I just use it directly on the BackpackUser model like this:

```
activity()
    ->performedOn($anEloquentModel)
    ->causedBy($user)
    ->withProperties(['customProperty' => 'customValue'])
    ->log('Look mum, I logged something');
```

I'm sure that it didn't log any activity before I put in this code.

The only weird thing I noticed is when I changed the relationship to hasOne or if I didn't select any value for user_type_id, it only logs it once. But when the relationship is belongsTo and the user_type_id has a value, it logs it twice.

**Reply by Gummibeer** [reply]
Are you able to provide a running repo with your app or at least the essential model setup?
If you can provide a repo with a Laravel app, your model setup that reproduces your issue I would be able to dig into it and check what happens, where the activities come from and why they end up twice in the DB.

Without it will get harder and I can only guess but you will have to do the real debug work.

At first I would try to get an idea of what happens by using `debug_backtrace()` and `dump()` or `dd()`. If you don't want to interrupt your output you can use `ld()` by https://github.com/spatie/laravel-log-dumper or any other dumper that doesn't dump into the respoonse.

**Reply by astarialexi** [reply]
Hi. Thanks so much. It wasn't activity log that was the problem.. We were overwriting the save function in the model and it's being called twice when the relationship is belongsTo.
